activation/handler_test.go

```diff
@@ -2040,107 +1321,104 @@ func TestHandler_AtxWeight(t *testing.T) {
-			Return(uint64(1), nil)
-		handler.mValidator.EXPECT().PositioningAtx(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any())
-		handler.mValidator.EXPECT().IsVerifyingFullPost().Return(true)
-
-		atx := newAtx(challenge, nipost, 2, types.Address{1, 2, 3, 4})
-		atx.SetValidity(types.Unknown)
-		require.NoError(t, SignAndFinalizeAtx(sig, atx))
-		_, proof, err := handler.SyntacticallyValidateDeps(context.Background(), atx)
+
+		atx := newActivationTxV1(t, goldenATXID)
+		atx.Sign(sig)
+
+		atxHdlr := newTestHandler(t, goldenATXID)
+		atxHdlr.expectAtxV1(atx, sig.NodeID(), func(o *atxHandleOpts) { o.distributedPost = false })
+		err = atxHdlr.HandleGossipAtx(context.Background(), "", codec.MustEncode(atx))
+
+		vatx, err := atxs.Get(atxHdlr.cdb, atx.ID())
 		require.NoError(t, err)
-		require.Nil(t, proof)
-		require.Equal(t, types.Valid, atx.Validity())
+		require.Equal(t, types.Valid, vatx.Validity())
 	})
-	t.Run("post verified fully", func(t *testing.T) {
+	t.Run("post not verified fully (distributed post)", func(t *testing.T) {
 		t.Parallel()
-		handler := newTestHandler(t, goldenATXID)
-		handler.mValidator.EXPECT().InitialNIPostChallenge(gomock.Any(), gomock.Any(), gomock.Any())
-		handler.mValidator.EXPECT().
-			NIPost(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
-			Return(uint64(1), nil)
-		handler.mValidator.EXPECT().PositioningAtx(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any())
-		handler.mValidator.EXPECT().IsVerifyingFullPost().Return(false)
-
-		atx := newAtx(challenge, nipost, 2, types.Address{1, 2, 3, 4})
-		atx.SetValidity(types.Unknown)
-		require.NoError(t, SignAndFinalizeAtx(sig, atx))
-		_, proof, err := handler.SyntacticallyValidateDeps(context.Background(), atx)
+
+		atx := newActivationTxV1(t, goldenATXID)
+		atx.Sign(sig)
+
+		atxHdlr := newTestHandler(t, goldenATXID)
+		atxHdlr.expectAtxV1(atx, sig.NodeID(), func(o *atxHandleOpts) { o.distributedPost = true })
+		err = atxHdlr.HandleGossipAtx(context.Background(), "", codec.MustEncode(atx))
+
+		vatx, err := atxs.Get(atxHdlr.cdb, atx.ID())
 		require.NoError(t, err)
-		require.Nil(t, proof)
-		require.Equal(t, types.Unknown, atx.Validity())
+		require.Equal(t, types.Unknown, vatx.Validity())
 	})
 	require.NoError(t, err)
 }
+
+func newActivationTxV1(
+	t testing.TB,
+	goldenATXID types.ATXID,
+	opts ...func(*wire.ActivationTxV1),
+) *wire.ActivationTxV1 {
+	t.Helper()
+	nonce := uint64(999)
+	poetRef := types.RandomHash()
+	atx := &wire.ActivationTxV1{
+		InnerActivationTxV1: wire.InnerActivationTxV1{
+			NIPostChallengeV1: wire.NIPostChallengeV1{
+				PrevATXID:        types.EmptyATXID,
+				PublishEpoch:     postGenesisEpoch,
+				PositioningATXID: goldenATXID,
+				CommitmentATXID:  &goldenATXID,
+				InitialPost:      &wire.PostV1{},
+			},
+			NIPost:   newNIPosV1tWithPoet(t, poetRef.Bytes()),
+			VRFNonce: &nonce,
```

Comment:
If an ATX has the same VRF nonce as the previous one, the `VRFNonce` field should be `nil`. Do we follow this convention in the tests? Need to make sure `nil` nonce works as expected.
Note that after #5791, the nonce value is always stored in SQLite `atxs.nonce` column, but still it will be `nil` inside the ATX blob if the prev ATX's nonce is the same.

Reply:
This helper creates "the first" ATX (empty previous ATX), which is why it always includes the nonce. There is another helper `newChainedActivationTxV1` to create subsequent ATXs (they have nil VRFNonce field).

> Note that after #5791, the nonce value is always stored in SQLite atxs.nonce column, but still it will be nil inside the ATX blob if the prev ATX's nonce is the same.

That's unfortunate because such two ATXs would have different `ID`. We need to make sure to always calculate the ID based on the original wire version (as in the blob).

Reply:
There's not enough info in the `atxs` table to calculate the `ID`, so the contents of the blob has to be used in any case to (re)calculate the ID value

Reply:
I would like to add here that if this function is only used to generate initial ATXs its name should reflect that, maybe `newInitialATXv1` instead of `newActivationTxV1`?